ingest/ledgerbackend/cloud_storage_backend.go

```diff
@@ -0,0 +1,210 @@
+	}
+
+	if ledgerRange.bounded {
+		_, err := csb.GetLedger(ctx, ledgerRange.to)
+		if err != nil {
+			return errors.Wrapf(err, "error getting ending ledger %d", ledgerRange.to)
+		}
+	}
+
+	return nil
+}
+
+// IsPrepared is a no-op for CloudStorageBackend.
+func (csb *CloudStorageBackend) IsPrepared(ctx context.Context, ledgerRange Range) (bool, error) {
+	return true, nil
+}
+
+// Close is a no-op for CloudStorageBackend.
+func (csb *CloudStorageBackend) Close() error {
+	return nil
+}
+
+// TODO: Should this function also be modified and added to support/datastore?
+// This function should be shared between ledger exporter and this legerbackend reader
+func GetObjectKeyFromSequenceNumber(ledgerSeq uint32, ledgersPerFile uint32, filesPerPartition uint32) (string, error) {
+	var objectKey string
+
+	if ledgersPerFile < 1 {
+		return "", errors.Errorf("Invalid ledgers per file (%d): must be at least 1", ledgersPerFile)
+	}
+
+	if filesPerPartition > 1 {
+		partitionSize := ledgersPerFile * filesPerPartition
+		partitionStart := (ledgerSeq / partitionSize) * partitionSize
+		partitionEnd := partitionStart + partitionSize - 1
+		objectKey = fmt.Sprintf("%d-%d/", partitionStart, partitionEnd)
+	}
+
+	fileStart := (ledgerSeq / ledgersPerFile) * ledgersPerFile
+	fileEnd := fileStart + ledgersPerFile - 1
+	objectKey += fmt.Sprintf("%d", fileStart)
+
+	// Multiple ledgers per file
+	if fileStart != fileEnd {
+		objectKey += fmt.Sprintf("-%d", fileEnd)
+	}
+	objectKey += fileSuffix
+
+	return objectKey, nil
+}
+
+func getLatestDirectory(directories []string) (string, error) {
+	var latestDirectory string
+	largestDirectoryLedger := 0
+
+	for _, dir := range directories {
+		// dir follows the format of "ledgers/<network>/<start>-<end>"
+		// Need to split the dir string to retrieve the <end> ledger value to get the latest directory
+		dirParts := strings.Split(dir, "/")
+		lastDirPart := dirParts[len(dirParts)-1]
+		parts := strings.Split(lastDirPart, "-")
+
+		if len(parts) == 2 {
+			upper, err := strconv.Atoi(parts[1])
+			if err != nil {
+				return "", errors.Wrapf(err, "failed getting latest directory %s", dir)
+			}
+
+			if upper > largestDirectoryLedger {
+				latestDirectory = dir
+				largestDirectoryLedger = upper
+			}
+		}
+	}
+
+	return latestDirectory, nil
+}
+
+func getLatestFileNameLedgerSequence(fileNames []string, directory string) (uint32, error) {
+	latestLedgerSequence := uint32(0)
+
+	for _, fileName := range fileNames {
+		// Trim file down to just the ledgerSequence
+		fileNameTrimExt := strings.TrimSuffix(fileName, ".xdr.gz")
+		fileNameTrimPath := strings.TrimPrefix(fileNameTrimExt, directory+"/")
+		ledgerSequence, err := strconv.ParseUint(fileNameTrimPath, 10, 32)
+		if err != nil {
+			return uint32(0), errors.Wrapf(err, "failed converting filename to uint32 %s", fileName)
+		}
+
+		if uint32(ledgerSequence) > latestLedgerSequence {
+			latestLedgerSequence = uint32(ledgerSequence)
+		}
```

Comment:
could use our [`ordered.Max[T]`](https://github.com/stellar/go/blob/master/support/ordered/math.go) here, e.g.

```suggestion
		latestLedgerSequence = ordered.Max(latestLedgerSequence, uint32(ledgerSequence))
```